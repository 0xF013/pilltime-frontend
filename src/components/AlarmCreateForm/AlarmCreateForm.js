import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';
import { create } from '../../redux/alarms';
import { connect } from 'react-redux';

export class AlarmCreateForm extends Component {

  onSubmit = async (data) => {
    await this.props.create(data);
  };

  render() {
    const { handleSubmit } = this.props;
    return (
      <form onSubmit={handleSubmit(this.onSubmit)}>
        <h2>Create alarm</h2>
        <div>
          <label htmlFor="ingestionTime">Ingestion time:</label>
          <br />
          <Field name="ingestionTime" component="input" type="text" placeholder="10:30" />
        </div>
        <div>
          <label htmlFor="pillName">Pill name:</label>
          <br />
          <Field name="pillName" component="input" type="text" placeholder="Alprazolam" />
        </div>
        <div>
          <label htmlFor="dosage">Dosage:</label>
          <br />
          <Field name="dosage" component="input" type="text" placeholder="0.5mg" />
        </div>
        <button type="submit">Submit</button>
      </form>
    );
  }
}

const DecoratedWithForm = reduxForm({ form: 'alarmCreateForm'})(AlarmCreateForm);
export default connect(undefined, { create })(DecoratedWithForm);