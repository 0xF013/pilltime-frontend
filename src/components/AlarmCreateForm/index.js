import AlarmCreateForm from './AlarmCreateForm';
export default AlarmCreateForm;