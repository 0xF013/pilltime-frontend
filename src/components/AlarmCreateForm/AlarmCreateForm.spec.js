import React from 'react';
import { AlarmCreateForm } from './AlarmCreateForm';
import { shallow } from 'enzyme';

describe('<AlarmCreateForm />', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(
    <AlarmCreateForm
      handleSubmit={() => {}}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches the create action on submit', () => {
    const create = jest.fn();
    const wrapper = shallow(
    <AlarmCreateForm
      handleSubmit={() => {}}
      create={ create }
    />);
    wrapper.instance().onSubmit({some: 'data'});
    expect(create).toHaveBeenCalledWith({some: 'data'});
  });
});