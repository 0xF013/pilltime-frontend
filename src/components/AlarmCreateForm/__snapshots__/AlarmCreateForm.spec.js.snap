exports[`<AlarmCreateForm /> matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    onSubmit={undefined}>
    <h2>
        Create alarm
    </h2>
    <div>
        <label
            htmlFor="ingestionTime">
            Ingestion time:
        </label>
        <br />
        <Field
            component="input"
            name="ingestionTime"
            placeholder="10:30"
            type="text" />
    </div>
    <div>
        <label
            htmlFor="pillName">
            Pill name:
        </label>
        <br />
        <Field
            component="input"
            name="pillName"
            placeholder="Alprazolam"
            type="text" />
    </div>
    <div>
        <label
            htmlFor="dosage">
            Dosage:
        </label>
        <br />
        <Field
            component="input"
            name="dosage"
            placeholder="0.5mg"
            type="text" />
    </div>
    <button
        type="submit">
        Submit
    </button>
</form>,
  "nodes": Array [
    <form
      onSubmit={undefined}>
      <h2>
            Create alarm
      </h2>
      <div>
            <label
                  htmlFor="ingestionTime">
                  Ingestion time:
            </label>
            <br />
            <Field
                  component="input"
                  name="ingestionTime"
                  placeholder="10:30"
                  type="text" />
      </div>
      <div>
            <label
                  htmlFor="pillName">
                  Pill name:
            </label>
            <br />
            <Field
                  component="input"
                  name="pillName"
                  placeholder="Alprazolam"
                  type="text" />
      </div>
      <div>
            <label
                  htmlFor="dosage">
                  Dosage:
            </label>
            <br />
            <Field
                  component="input"
                  name="dosage"
                  placeholder="0.5mg"
                  type="text" />
      </div>
      <button
            type="submit">
            Submit
      </button>
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AlarmCreateForm
        handleSubmit={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AlarmCreateForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onSubmit": [Function],
        "props": Object {
          "handleSubmit": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          onSubmit={undefined}>
          <h2>
                    Create alarm
          </h2>
          <div>
                    <label
                              htmlFor="ingestionTime">
                              Ingestion time:
                    </label>
                    <br />
                    <Field
                              component="input"
                              name="ingestionTime"
                              placeholder="10:30"
                              type="text" />
          </div>
          <div>
                    <label
                              htmlFor="pillName">
                              Pill name:
                    </label>
                    <br />
                    <Field
                              component="input"
                              name="pillName"
                              placeholder="Alprazolam"
                              type="text" />
          </div>
          <div>
                    <label
                              htmlFor="dosage">
                              Dosage:
                    </label>
                    <br />
                    <Field
                              component="input"
                              name="dosage"
                              placeholder="0.5mg"
                              type="text" />
          </div>
          <button
                    type="submit">
                    Submit
          </button>
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          onSubmit={undefined}>
          <h2>
                    Create alarm
          </h2>
          <div>
                    <label
                              htmlFor="ingestionTime">
                              Ingestion time:
                    </label>
                    <br />
                    <Field
                              component="input"
                              name="ingestionTime"
                              placeholder="10:30"
                              type="text" />
          </div>
          <div>
                    <label
                              htmlFor="pillName">
                              Pill name:
                    </label>
                    <br />
                    <Field
                              component="input"
                              name="pillName"
                              placeholder="Alprazolam"
                              type="text" />
          </div>
          <div>
                    <label
                              htmlFor="dosage">
                              Dosage:
                    </label>
                    <br />
                    <Field
                              component="input"
                              name="dosage"
                              placeholder="0.5mg"
                              type="text" />
          </div>
          <button
                    type="submit">
                    Submit
          </button>
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AlarmCreateForm
    handleSubmit={[Function]} />,
}
`;
